Sample/Unity Sample Application/Assets/TitleGUI.cs

```diff
@@ -42,67 +57,56 @@ public class TitleGUI : MonoBehaviour {
-
-		GUI.enabled = true;
-
-		GUILayout.Label ("AppID " + appID, titleLabelStyle);
-		if (GUILayout.Button ("Init SDK")) {
-			string[] array = new string[placements.Keys.Count];
-			placements.Keys.CopyTo(array, 0);
-			Vungle.init (appID, appID, appID, array);
-			initializeEventHandlers ();
-		}
-
-		GUI.enabled = adInited;
-		GUILayout.Label ("Placement 1");
-		GUILayout.Label ("PlacementID " + list[0]);
-		GUI.enabled = placements[list[0]];
-		if (GUILayout.Button ("play")) {
-			Vungle.playAd(list[0]);
-		}
-		GUI.enabled = adInited;
-
-		GUILayout.Label ("Placement 2");
-		GUILayout.Label ("PlacementID " + list[1]);
-		GUI.enabled = placements[list[1]];
-		GUILayout.BeginHorizontal ();
-		if (GUILayout.Button ("play")) {
-			Vungle.playAd(list[1]);
-		}
-		GUI.enabled = !placements[list[1]];
-		if (GUILayout.Button ("load")) {
-			Vungle.loadAd(list[1]);
-		}
-		GUILayout.EndHorizontal();
-		GUI.enabled = adInited;
-
-		GUILayout.Label ("Placement 3");
-		GUILayout.Label ("PlacementID " + list[2]);
-		GUI.enabled = placements[list[2]];
-		GUILayout.BeginHorizontal ();
-		if (GUILayout.Button ("play")) {
-			Vungle.playAd(list[2]);
-		}
-		GUI.enabled = !placements[list[2]];
-		if (GUILayout.Button ("load")) {
-			Vungle.loadAd(list[2]);
-		}
-		GUILayout.EndHorizontal();
-
-		GUILayout.EndArea ();
+
+	void SetupButtonsAndText () {
+		placementIdList = new List<string>(placements.Keys);
+
+		appIDText.text = "App ID: " + appID;
+		placementID1Text.text = "Placement ID: " + placementIdList [0]; 
+		placementID2Text.text = "Placement ID: " + placementIdList [1]; 
+		placementID3Text.text = "Placement ID: " + placementIdList [2]; 
+
+		initButton.onClick.AddListener (onInitButton);
+		playPlacement1Button.onClick.AddListener (onPlayPlacement1);
+		loadPlacement2Button.onClick.AddListener (onLoadPlacement2);
+		playPlacement2Button.onClick.AddListener (onPlayPlacement2);
+		loadPlacement3Button.onClick.AddListener (onLoadPlacement3);
+		playPlacement3Button.onClick.AddListener (onPlayPlacement3);
+	}
+
+	void onInitButton() {
+		string[] array = new string[placements.Keys.Count];
+		placements.Keys.CopyTo(array, 0);
+		Vungle.init (appID, appID, appID, array);
+		initializeEventHandlers ();
+	}
+
+	void onPlayPlacement1 () {
+		Vungle.playAd(placementIdList[0]);
+	}
+
+	void onLoadPlacement2 () {
+		Vungle.loadAd(placementIdList[1]);
+	}
+
+	void onPlayPlacement2 () {
+		Vungle.playAd(placementIdList[1]);
+	}
+
+	void onLoadPlacement3 () {
+		Vungle.loadAd(placementIdList[2]);
+	}
+
+	void onPlayPlacement3 () {
+		Vungle.playAd(placementIdList[2]);
 	}
 
-	void initializeGUIStyles() {
-		titleLabelStyle = new GUIStyle ();
-		titleLabelStyle.stretchWidth = true;
-//		titleLabelStyle.stretchHeight = true;
-		titleLabelStyle.fixedWidth = Screen.width;
-//		titleLabelStyle.fixedHeight = headerHeight;
-		titleLabelStyle.alignment = TextAnchor.MiddleCenter;
+	void updateButtonState() {
+		playPlacement1Button.enabled = placements[placementIdList[0]];
```

Comment:
NullReferenceException: Object reference not set to an instance of an object
TitleGUI.SetupButtonsAndText () (at Assets/TitleGUI.cs:64)
TitleGUI.Start () (at Assets/TitleGUI.cs:45)